ember-prismic-dom/src/components/prismic/dom.ts

```diff
@@ -0,0 +1,73 @@
+import Component from '@glimmer/component';
+
+import { asTree } from '@prismicio/richtext';
+
+import type { ComponentLike } from '@glint/template';
+import type { RTNode } from '@prismicio/types';
+import type { Tree, TreeNode } from '../../unpublished-development-types';
+
+export interface PrismicDomArgs {
+  [key: string]:
+    | ComponentLike
+    | RTNode[]
+    | string
+    | ((node: TreeNode) => void)
+    | undefined;
+  nodes?: RTNode[] | string;
+  onUnknownTag?: (node: TreeNode) => void;
+}
+
+export interface PrismicDomSignature {
+  Element: HTMLDivElement;
+  Args: PrismicDomArgs;
+}
+
+/**
+ * A component that renders a rich text document from Prismic data.
+ * @example
+ * ```hbs
+ * <Prismic::Dom @nodes={{@myPrismicDoc.data.myRichText}} />
+ * ```
+ */
+export default class PrismicDom extends Component<PrismicDomSignature> {
+  /**
+   * Gets the tree representation of the rich text data.
+   * @returns The tree representation of the rich text data.
+   * @type {Tree}
+   */
+  get tree(): Tree {
+    const nodes = (this.args.nodes || []) as RTNode[];
+
+    return asTree(nodes);
+  }
+
+  /**
+   * Checks if `nodes` is a string.
+   * @returns A boolean indicating whether `nodes` is of type string.
+   * @type {boolean}
+   */
+  get isString(): boolean {
+    return typeof this.args.nodes === 'string';
+  }
+
+  /**
+   * Gets the custom component mappings.
+   * @returns An object that maps custom component names to their component definitions.
+   * @type {Record<string, ComponentLike>}
+   */
+  get componentNames(): Record<string, ComponentLike> {
```

Comment:
🟢 _Feel free to resolve_

Do you think we could abstract the type `Record<string, ComponentLike>`?

Reply:
@vscav yes, good idea. WDY about `export type CustomComponentMap = Record<string, CustomComponentLike>;`?